medicines/api/src/product.rs

```diff
@@ -33,26 +34,19 @@ pub fn handle_doc(document: &IndexResult, products: &mut Vec<Product>) {
     }
 }
 
-pub async fn get_products_by_substance_name(
-    substance_name: &str,
-    client: &impl Search,
-) -> Result<Vec<Product>, reqwest::Error> {
-    // Get a list of documents from Azure which are about products containing the
-    // substance name.
+pub async fn get_substance_with_products(name: &str, client: &impl Search) -> Result<Substance, reqwest::Error> {
```

Comment:
I think it'd be good to stick with `substance_name` here, just makes it that bit more explicit since both products and substances are in play here.